skyportal/handlers/api/shift.py

```diff
@@ -136,32 +136,66 @@ def get(self, group_id=None):
                 schema: Error
         """
         if group_id is not None:
-            shifts = (
+            queried_shifts = (
                 Shift.query_records_accessible_by(
                     self.current_user,
                     mode="read",
                     options=[
-                        joinedload(Shift.group).joinedload(Group.users),
-                        joinedload(Shift.users),
+                        joinedload(Shift.group).joinedload(Group.group_users),
+                        joinedload(Shift.shift_users),
                     ],
                 )
                 .filter(Shift.group_id == group_id)
                 .order_by(Shift.start_date.asc())
                 .all()
             )
         else:
-            shifts = (
+            queried_shifts = (
                 Shift.query_records_accessible_by(
                     self.current_user,
                     mode="read",
                     options=[
-                        joinedload(Shift.group).joinedload(Group.users),
-                        joinedload(Shift.users),
+                        joinedload(Shift.group).joinedload(Group.group_users),
+                        joinedload(Shift.shift_users),
                     ],
                 )
                 .order_by(Shift.start_date.asc())
                 .all()
             )
+        shifts = []
+        for shift in queried_shifts:
+            susers = [
+                {
+                    "id": su.user.id,
+                    "username": su.user.username,
+                    "first_name": su.user.first_name,
+                    "last_name": su.user.last_name,
+                    "contact_email": su.user.contact_email,
+                    "contact_phone": su.user.contact_phone,
+                    "admin": su.admin,
+                }
+                for su in shift.shift_users
+            ]
+            gusers = [
+                {
+                    "id": gu.user.id,
+                    "username": gu.user.username,
+                    "first_name": gu.user.first_name,
+                    "last_name": gu.user.last_name,
+                    "contact_email": gu.user.contact_email,
+                    "contact_phone": gu.user.contact_phone,
+                    "admin": gu.admin,
+                }
+                for gu in shift.group.group_users
+            ]
+            shift = shift.to_dict()
+            shift["users"] = susers
+            shift.pop('shift_users', [])
```

Comment:
What is this doing?

Reply:
Hey @mcoughlin , the `shift` is not a dict but a `Shift` object, and i can't assign a new values to it, so i need to convert it in a dict. Then i set the `users` with the new `susers` i created, and i get rid of `shift_users` that is useless now that i have a `users` key. I could just set the `shift_users` to `susers`. I did that because i used the `users` everywhere in the frontend, but if you want me to avoir using pop, i can just replace `users` by `shift_users` in the frontend

Reply:
Is that a painful change? If it is more direct, it seems cleaner.

Reply:
Ok ! I'll do that tomorrow then. Have a nice day Michael